packages/toolpad-app/src/toolpad/Apps/index.tsx

```diff
@@ -728,6 +729,18 @@ export default function Home() {
 
   const AppsView = viewMode === 'list' ? AppsListView : AppsGridView;
 
+  const [searchText, setSearchText] = React.useState('');
+
+  const handleSearchInput = React.useCallback((event: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchText(event.target.value);
+  }, []);
+
+  const filteredApps = React.useMemo(() => {
```

Comment:
I think this works well, but as a short-term solution - long term we should have pagination when loading a list of apps, and the search should filter the results server-side.